ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -253,9 +241,7 @@ impl Module {
     ///
     /// If `args` would overflow the index type.
     fn push_args(&mut self, args: Vec<Operand>) -> Result<ArgsIdx, CompilationError> {
-        let idx = self.args.len();
-        self.args.extend(args);
-        ArgsIdx::new(idx)
+        ArgsIdx::new(self.args.len()).inspect(|_| self.args.extend(args))
```

Comment:
This may be shorter, but I find it more difficult to think about. The old version is quite clear: Extend args, but return the previous length. In this new version I find this harder to see. But I'm not going to die on this hill if you think this is still better.

Reply:
The downside to this change is that you need to know what `inspect` does, but it's a useful function, and I think the terseness is just about worth it on balance.

Reply:
As I said, I don't feel super strongly about it.

Reply:
I forgot to say: there is an advantage to the new versions. It only mutates the self struct if the index is representable. In the old version, you got back a CompilationError but the struct had been mutated. So in that small way, the new version is functionally different (and IMHO better).

Reply:
I did notice that, but thought since in those cases we have to abort the trace it doesn't matter too much that the vec was already changed. But fundamentally you're right.

Reply:
I'm struggling to understand this. Where is `inspect()` defined?

Reply:
`Result::inspect` in the standard library.

Reply:
Huh. Clever, but strange :)